### PR TITLE
More options to organization scope mapper including adding organization attributes to tokens

### LINKED ISSUE #31642: Include organization attributes and information in ID and access tokens
### Description

It would be beneficial to enhance the token payload to include organization-specific attributes and additional information. This will allow easier access to organization data on the client side without requiring additional API calls.
Specifically:

- Organization attributes
- Additional organization information
- Keycloak ID of the organization

Proposed token structure:

```
{
  "organization": {
    "id": "keycloak-org-id",
    "name": "myOrg",
    "attributes": {
      "key1": "value1",
      "key2": "value2"
    }
  }
}
```

This change should be implemented for both ID tokens and access tokens.
Benefits:

- Improved performance by reducing the need for separate API calls to fetch organization data
- Enhanced security by including organization context in the token
- Easier development of organization-aware application

### Discussion

#30180

### Motivation

_No response_

### Details

_No response_

### PR DESCRIPTION
**Pull request by pedroigor**
Closes #31642

* Adds a setting to add organization attributes when mapping organizations to tokens
* Adds setting to map single or multiple organizations based on the `Multivalued`
* Adds setting to choose if the `organization` claim should be a list (string values) or a JSON object.
* By default, the `organization` claim will be a list of organization aliases.

### CODE REVIEW

**Comment by pedroigor** [comment]
@Degoya-Studio Can you check it out?

**Comment by Degoya-Studio** [comment]
Hi @pedroigor I've check out your branch and started keycloak with:

`java -jar quarkus/server/target/lib/quarkus-run.jar start-dev --features organization
2024-08-25 12:37:39,743 INFO  [org.keycloak.common.Profile] (main) Preview features enabled: organization:v1`

However, I don't see any option to enable or add organizations in the user interface. Am I missing something?

**Comment by ahus1** [comment]
Last time I tested organisations, it seemed to me I needed first create a new realm (not the master realm). There I was able to enable it in the realm tab.

**Comment by Degoya-Studio** [comment]
@ahus1 I've done that, but I still don't see the option.
Keycloak

**Comment by pedroigor** [comment]
@Degoya-Studio Did you create a new realm?

**Review by sguilhen — APPROVED** [review]

**Review by ahus1 — APPROVED** [review]
Approving based on @sguilhen's review. 

Not merging yet in case you want to wait for community feedback.

**Comment by pedroigor** [comment]
@Degoya-Studio The screenshot is also missing the `Organizations` switch at the realm settings. I suspect the feature was not enabled when starting the server.

**Comment by Degoya-Studio** [comment]
Hi @pedroigor this is how I started the server

`java -jar quarkus/server/target/lib/quarkus-run.jar start-dev --features="organization"`

The console returned

> 2024-08-27 01:10:06,735 INFO  [org.keycloak.common.Profile] (main) Preview features enabled: organization:v1

I then created a realm "**dzg**" and created a new user as realm-admin
Bildschirmfoto 2024-08-27 um 00 57 50

However, when I go on the realm settings, I still don't see the switch, that I would normally see on the main branch of Keycloak.

Bildschirmfoto 2024-08-27 um 00 57 58

I don't know what I'm missing here but I guess it's fine with me. I don't think I will have this issue once the branch is merged.

**Review by edewit — APPROVED** [review]

**Pull request merged**
